### DISCUSSION TITLE
Unable to add devices

### DISCUSSION THREAD

**Discussion by Th3M1k3y** [Koenkk/zigbee2mqtt#10829] [Devices]
I got a Sonoff Zigbee 3.0 dongle, and using these versions
```
Zigbee2MQTT version: 1.22.2 commit: 1a0a9a6e
Coordinator type: zStack3x0
Coordinator revision: 20211217
```
I started out by wanting to add this one to the network https://zigbee.blakadder.com/Neo_NAS-WR01B.html but failed getting it to join. After that I then tried to flash my old xiaomi gateway with a custom firmware to allow it to act as a router, that too failed to join the network. After that I tried with two new Xiaomi buttons, they too failed to get onto the network. Last I then tried to remove a device already working, but re-adding it to the network also fails.

With log level set to debug I am getting no useful information, only that it is allowing new devices to join, and then messages from the working devices.

I don't know how to proceed. I tried both updating the coordinator firmware to the latest, and also zigbee2mqtt to latest version. I have spend over 3 hours trying to figure out what is going on, but I don't see any log messages or anything pointing me in any direction.

**Comment by Th3M1k3y** [comment]
Got it working, I had to remove the powered usb hdd also connected.

**Reply by jrhansen** [reply]
I think I see a very similar issue.

Do you happen to be using a Raspberry Pi 4?

I have an external HDD (with external power supply) connected to my RPi 4, and it seems that removing it and booting from some USB flash drive instead, makes the Sonoff dongle work (with an USB extension cable). The HDD that you had to remove, does that also have external power supply or is it powered from the USB port?

**Reply by Th3M1k3y** [reply]
> Do you happen to be using a Raspberry Pi 4?

Yes

> I have an external HDD (with external power supply) connected to my RPi 4, and it seems that removing it and booting from some USB flash drive instead, makes the Sonoff dongle work (with an USB extension cable). The HDD that you had to remove, does that also have external power supply or is it powered from the USB port?

The HDD was a WD Elements with external power supply.

**Reply by jrhansen** [reply]
Actually very close to my own setup. If you are interested I got some very interesting feedback regarding this issue in another thread (#11159).

Thanks for your reply.
